### PR TITLE
Use net.Buffers in Buffer.DumpTo

### PR DESCRIPTION
**Pull request by CAFxX**
Use [`net.Buffers`](https://golang.org/pkg/net/#Buffers) when writing the contents of a Buffer to a Writer. This allows to use `writev` on most platforms when the writer is a network connection.

### CODE REVIEW

File: buffer/pool.go [hunk 1]
```diff
@@ -156,26 +155,22 @@ func (b *Buffer) Size() int {
...
 // DumpTo outputs the contents of a buffer to a writer and resets the buffer.
 func (b *Buffer) DumpTo(w io.Writer) (written int, err error) {
-	var n int
-	for _, buf := range b.bufs {
-		if err == nil {
-			n, err = w.Write(buf)
-			written += n
-		}
-		putBuf(buf)
+	bufs := (net.Buffers)(b.bufs)
```
Comment (GoWebProd) [inline buffer/pool.go]: Maybe change b.bufs type to net.Buffers?

File: buffer/pool.go [hunk 2]
```diff
@@ -156,26 +155,22 @@ func (b *Buffer) Size() int {
...
+		putBuf(buf)
 	}
 	putBuf(b.toPool)
 
 	b.bufs = nil
 	b.Buf = nil
 	b.toPool = nil
 
-	return
+	return (int)(wr), err
```
Comment (GoWebProd) [inline buffer/pool.go]: int(wr) - more canonical

File: buffer/pool.go [hunk 3]
```diff
@@ -156,26 +155,22 @@ func (b *Buffer) Size() int {
...
-		if err == nil {
-			n, err = w.Write(buf)
-			written += n
-		}
-		putBuf(buf)
+	bufs := (net.Buffers)(b.bufs)
+	if len(b.Buf) > 0 {
+		bufs = append(bufs, b.Buf)
 	}
+	wr, err := bufs.WriteTo(w)
```
Comment (GoWebProd) [inline buffer/pool.go]: Maybe more clear name for `wr` or n as by default?

File: buffer/pool.go (hunk 1 above)
Comment (CAFxX) [reply buffer/pool.go]: I did not want to have this change impact unrelated parts of the code, and I could not see a benefit in changing it (compared to simply casting it).

If you really prefer it I can do it, I don't think it should be hard to change.

**Pull request merged**